### PR TITLE
Musl support

### PR DESCRIPTION
**Pull request by gnurizen**
Add support for pulling unwind info from debug link that is available
locally, if debug link can't be found dwarf unwinding will fail. Future work
may have the agent to pull debuginfo from parca server and allow
dwarf unwinding w/o debug link files available locally.

If the debug link file doesn't match the CRC report an error and don't
use it.

Fix finder so that it doesn't return debuglink matches pointing back at
the exe itself, this just leads to spurious crc failures.

### CODE REVIEW

File: pkg/objectfile/object_file.go
Comment (kakkoyun) [inline pkg/objectfile/object_file.go:33]: Why do we export this?

File: pkg/stack/unwind/unwind_table.go
Comment (kakkoyun) [inline pkg/stack/unwind/unwind_table.go:206]: It worth checking our debuginfo package. Especially the `Find`. I believe we can reuse some code pieces.

File: pkg/objectfile/object_file.go
Comment (gnurizen) [reply pkg/objectfile/object_file.go]: Ignore, WIP temporary hack.

File: pkg/stack/unwind/unwind_table.go
Comment (gnurizen) [reply pkg/stack/unwind/unwind_table.go:227]: Thanks for the tip!  I especially want to nail the CRC checking before landing this.

**Comment by gnurizen** [comment]
@umanwizard  can you review this?   Thanks!

**Review by brancz — APPROVED** [review]

File: bpf/unwinders/native.bpf.c
Comment (gnurizen) [reply bpf/unwinders/native.bpf.c:695]: Will do

File: bpf/unwinders/native.bpf.c
Comment (gnurizen) [reply bpf/unwinders/native.bpf.c:1414]: Too verbose

File: pkg/stack/unwind/compact_unwind_table.go
Comment (gnurizen) [reply pkg/stack/unwind/compact_unwind_table.go]: I tried to repro it and couldn't, I wonder if it was from parsing .debug_frame wrong.   We'll see what more testing shows.

File: pkg/stack/unwind/unwind_table.go
Comment (gnurizen) [reply pkg/stack/unwind/unwind_table.go:273]: Well we should have a legit context here but we don't, this is a bread crumb to remember to wire in a real context.

**Comment by umanwizard** [comment]
I still don’t have approval rights, but it looks good to me overall, so merge at your discretion!

File: pkg/stack/unwind/unwind_table.go
Comment (gnurizen) [reply pkg/stack/unwind/unwind_table.go:276]: Sure

File: pkg/stack/unwind/unwind_table.go
Comment (gnurizen) [reply pkg/stack/unwind/unwind_table.go:291]: Nice catch!

File: bpf/unwinders/native.bpf.c
Comment (gnurizen) [reply bpf/unwinders/native.bpf.c:695]: It seems to work on ubuntu 20.04 LTS (5.14 kernel).  Are you sure you weren't just fixing printfs with more than 3 args?

**Pull request merged**
